### ISSUE TITLE
Weighted loss from conditional random field

### ISSUE THREAD

**Issue by tavo91** [allenai/allennlp#2526]
I was wondering if there's any plan on providing weighted loss from the conditional random field module. For example, with `nn.CrossEntropyLoss` it is possible to pass `class_weights` that are applied to specific labels. This is very useful when your data is unbalanced.

**Describe the solution you'd like**
I was taking a look at how the implementation would go if that were to be added. And, based on this paper, it seems that the difficult part lies in the normalizer term `log(Z(x))`:
https://perso.uclouvain.be/michel.verleysen/papers/ieeetbe12gdl.pdf

They mention two possible ways to calculate `log(Z(x))`. One is to compute the weighted terms at time t in the log space (which becomes a bit tangled), and the other one is more practical: follow a scaling trick that normalizes alphas_k at timestep t using the sum of the alphas_k (see Equation 16).

I will be trying the implementation of the scaling trick, but just wanted to know if this is on the road for new features.

**Comment by schmmd** [comment]
We have no plans to do this.  We would accept a contribution but this seems quite niche, so I'm closing this issue for now.  Please comment if you'd like us to re-open.
